src/main.rs

```diff
@@ -298,10 +308,167 @@ async fn create_endpoint(
     if let Some(addr) = common.ipv6_addr {
         builder = builder.bind_addr_v6(addr);
     }
+    if common.persist || common.persist_at.is_some() {
+        builder = builder.secret_key(get_secret_key(common.persist_at.as_ref()).await);
+    }
     let endpoint = builder.bind().await?;
     Ok(endpoint)
 }
 
+async fn get_secret_key(persist_at: Option<&PathBuf>) -> SecretKey {
+    let persist_at_cow = persist_at
+        .map(Cow::from) // Reference
+        .or_else(|| {
+            std::env::home_dir().map(|mut p| {
+                p.push(".auth");
```

Comment:
shouldn't this be stored in a platform-specific app folder?
eg. `dirs::config_dir().join("dumbpipe").join("dumbpipe.key")`
with https://crates.io/crates/dirs